xonsh/built_ins.py

```diff
@@ -949,6 +949,10 @@ def run_subproc(cmds, captured=False, envs=None):
     if builtins.__xonsh__.env.get("XONSH_TRACE_SUBPROC"):
         print("TRACE SUBPROC: %s" % str(cmds), file=sys.stderr)
 
+    precmds = events.on_subproc_precommand.fire(cmds=cmds)
+    if precmds:
+        cmds = precmds[0]        
```

Comment:
Shouldn't this be `cmds =  precmds`?

Reply:
`precmds` will be `[(['ls',])]` after firing. So we need `[0]`.

Reply:
Why is it returning cmds wrapped in list?

Reply:
Because
https://github.com/anki-code/xonsh/blob/patch-2/xonsh/events.py#L178-L186

Reply:
Ahh right, then why is this only taking the first event's response? It seems like it should figure out what to do if multiple handlers are registered to this event

Reply:
Good question and I have no answer :) I suggest to use the first event and someone who will face with issues change this logic along his use case.

Reply:
Hmm we should probably figure this out